File: src/main/scala/piles/Lane.scala
package piles

import deck.Card
import deck.Rank
import java.util


class Lane(cards: Seq[Card]) extends Pile(cards) with Iterable[Card] {
  def isCardValid(card: Card): Boolean = {
    var valid = false
    valid = getTopCard match
      case Some(topCard) => (topCard.color ne card.color) && topCard.isNext(card)
      case None => card.rank eq Rank.KING
    valid
  }
}

File: src/main/old/piles/Lane.java
package piles;

import java.util.List;

public class Lane extends Pile implements Iterable<Card>{
    public Lane(List<Card> cards){
        super(cards);
    }

    public boolean isCardValid(Card card) {
        boolean valid = false;
        if(getTopCard() != null){
            valid = getTopCard().getColor() != card.getColor() &&
                    getTopCard().isNext(card);
        }else if(card.getRank()== Rank.KING){
            valid=true;
        }
        return valid;
    }
}
